homeassistant/components/smlight/coordinator.py

```diff
@@ -42,12 +52,21 @@ def __init__(self, hass: HomeAssistant, host: str) -> None:
         )
 
         self.unique_id: str | None = None
-        self.client = Api2(host=host, session=async_get_clientsession(hass))
         self.legacy_api: int = 0
 
+        if not hasattr(self, "client"):
+            self.client = self._init_client(host)
+
+        # self.client = SmDataUpdateCoordinator.client
+
+    def _init_client(self, host: str) -> Api2:
+        """Initialize the SMLIGHT client."""
+        client = Api2(host=host, session=async_get_clientsession(self.hass))
```

Comment:
Have you considered just creating the object in init.py and passing it to the coordinators separately